lib/Sema/TypeCheckPattern.cpp

```diff
@@ -123,16 +188,90 @@ lookupUnqualifiedEnumMemberElement(TypeChecker &TC, DeclContext *DC,
 }
 
 /// Find an enum element in an enum type.
-static EnumElementDecl *
-lookupEnumMemberElement(TypeChecker &TC, DeclContext *DC, Type ty,
-                        Identifier name, SourceLoc UseLoc) {
+static EnumElementDecl *lookupEnumMemberElement(TypeChecker &TC,
+                                                DeclContext *DC, Type ty,
+                                                DeclName name,
+                                                SourceLoc UseLoc) {
   if (!ty->mayHaveMembers())
     return nullptr;
 
   // Look up the case inside the enum.
   // FIXME: We should be able to tell if this is a private lookup.
   NameLookupOptions lookupOptions = defaultMemberLookupOptions;
   LookupResult foundElements = TC.lookupMember(DC, ty, name, lookupOptions);
+
+  ArrayRef<Identifier> queryParams = name.getArgumentNames();
+
+  if (foundElements.size() != 1) {
+    // Look up enum element only with base name
+    foundElements =
+        TC.lookupMember(DC, ty, name.getBaseIdentifier(), lookupOptions);
+
+    int numFoundEntries = 0;
+    LookupResultEntry hold = nullptr;
+    LookupResult singleResult;
+    for (LookupResultEntry resultEnt : foundElements) {
+      ValueDecl *e = resultEnt.getValueDecl();
+      if (auto *var = dyn_cast<VarDecl>(e)) {
```

Comment:
Since you're not using `var`, you can simply do an `isa` check.

```suggestion
      if (isa<VarDecl>(e)) {
```